pokemongo_bot/cell_workers/move_to_map_pokemon.py

```diff
@@ -0,0 +1,184 @@
+                pokemon['longitude'],
+            )
+
+            if pokemon['dist'] > self.config['max_distance'] and not self.config['snipe']:
+                continue
+
+            pokemon_list.append(pokemon)
+
+        return pokemon_list
+
+    def add_caught(self, pokemon):
+        for caught_pokemon in self.caught:
+            if caught_pokemon['encounter_id'] == pokemon['encounter_id']:
+                return
+        if len(self.caught) >= 200:
+            self.caught.pop(0)
+        self.caught.append(pokemon)
+
+    def was_caught(self, pokemon):
+        for caught_pokemon in self.caught:
+            if pokemon['encounter_id'] == caught_pokemon['encounter_id']:
+                return True
+        return False
+
+    def update_map_location(self):
+        if not self.config['update_map']:
+            return
+        try:
+            req = requests.get('{}/loc'.format(self.config['address']))
+        except requests.exceptions.ConnectionError:
+            logger.log('Could not reach PokemonGo-Map Server', color='red')
+            return
+        loc_json = req.json()
+
+        dist = distance(
+            self.bot.position[0],
+            self.bot.position[1],
+            loc_json['lat'],
+            loc_json['lng']
+        )
+
+        # update map when 500m away from center and last update longer than 2 minutes away
+        now = int(time.time())
+        if dist > 500 and now - self.last_map_update > 2 * 60:
+            requests.post('{}/next_loc?lat={}&lon={}'.format(self.config['address'], self.bot.position[0], self.bot.position[1]))
+            logger.log('Updated PokemonGo-Map position')
+            self.last_map_update = now
+
+    def snipe(self, pokemon):
+        last_position = self.bot.position[0:2]
+        self.bot.check_session(last_position)
+        self.bot.heartbeat()
+
+        logger.log('Teleporting to {} ({})'.format(pokemon['name'], format_dist(pokemon['dist'], self.unit)), 'green')
+        self.bot.api.set_position(pokemon['latitude'], pokemon['longitude'], 0)
+
+        logger.log('Encounter pokemon', 'green')
+        catch_worker = PokemonCatchWorker(pokemon, self.bot)
+        api_encounter_response = catch_worker.create_encounter_api_call()
+
+        time.sleep(2)
+        logger.log('Teleporting back to previous location..', 'green')
+        self.bot.api.set_position(last_position[0], last_position[1], 0)
+        time.sleep(2)
+        self.bot.heartbeat()
+
+        catch_worker.work(api_encounter_response)
+        self.add_caught(pokemon)
+
+        return WorkerResult.SUCCESS
+
+    def dump_caught_pokemon(self):
+        user_data_map_caught = 'data/map-caught-{}.json'.format(self.bot.config.username)
+        with open(user_data_map_caught, 'w') as outfile:
+            json.dump(self.caught, outfile)
+
+    def work(self):
+        # check for pokeballs (excluding masterball)
+        pokeballs = 0
+        pokeballs += self.bot.item_inventory_count(1)
```

Comment:
yeah `item_inventory_count` returned None instead of 0 -.- fixed it with 6ee5ea5